packages/venia-ui/lib/components/MiniCart/miniCart.gql.js

```diff
@@ -0,0 +1,37 @@
+import gql from 'graphql-tag';
+
+import { ProductListFragment } from './ProductList/productList.gql';
+
+export const MINI_CART_QUERY = gql`
+    query MiniCartQuery($cartId: String!) {
+        cart(cart_id: $cartId) @connection(key: "Cart") {
+            id
+            total_quantity
+            ...ProductListFragment
```

Comment:
Would be good to pull these out into a `MiniCartFragment` that we can then import for the `addProductToCart` mutation response.

Reply:
Can you elaborate? I didn't understand what you meant here.

Reply:
I'm just suggesting we try to use composition where possible for the queries/fragments. It removes the guess work for other components that need to update global things. For example:

```gql
mutation AddToCart(args) {
    cart {
       ...MiniCartFragment
    }
}
```

`MiniCartFragment` would just be `id`, `total_quantity` and include `...ProductListFragment`.

Reply:
I have created the `MiniCartFragment` in 018543e. At first, I did not understand the use of it but kind of made sense when I took a look at https://github.com/magento/pwa-studio/blob/develop/packages/venia-ui/lib/components/CartPage/cartPageFragments.gql.js#L8. I am not sure if that is what you meant when you said removing the guesswork for components.

Reply:
It's just one of those pre-optimizations so we don't have to keep the fragments in sync whenever we update. For example, if we later add some display data to the mini cart we would need to make sure it also gets fetched when we add to cart. Now, add to cart can just import the mini cart fragment and it'll automatically work when we update the fragment. Without this we would have to update multiple gql queries.